.github/workflows/tests.yml

```diff
@@ -40,10 +40,44 @@ jobs:
         if: github.event_name == 'pull_request'
         id: formulae-detect
 
+  timeout_labels:
+    if: github.event_name == 'pull_request' && github.repository == 'Homebrew/homebrew-core'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Count pull requests with the `CI-long-timeout` label
+        id: count_long_timeout_prs
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const query = `query($owner:String!, $name:String!, $label:String!) {
+              repository(owner:$owner, name:$name) {
+                pullRequests(last:100, states:OPEN, labels: [$label]) {
+                  totalCount
+                }
+              }
+            }`;
+            const variables = {
+              owner: context.repo.owner,
+              name: context.repo.repo,
+              label: 'CI-long-timeout'
+            }
+            const { repository: { pullRequests: { totalCount } } }  = await github.graphql(query, variables)
+            core.setOutput('count', totalCount)
+
+      - name: Remove `CI-long-timeout` label if applicable
+        if: fromJson(steps.count_long_timeout_prs.outputs.count) >= 2
+        uses: Homebrew/actions/label-pull-requests@master
+        with:
+          token: ${{secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN}}
```

Comment:
Ah, whoops. I don't think this token is accessible here. But then this probably needs a token with write access to pull requests.

Reply:
One way to get around this is to skip removing the label and just refusing to set a long timeout if there are too many pull requests with the label, but that's bound to make using the label very confusing.